spec/facter/resolvers/ec2_spec.rb

```diff
@@ -94,6 +94,12 @@
           }
         )
       end
+
+      it "uses the expected header" do
+        allow(Facter::Resolvers::Ec2).to receive(:v2_token).and_return(token)
+        expect(Facter::Util::Resolvers::Http).to receive(:get_request).with(metadata_uri, expected_header, any_args)
```

Comment:
There should be a spy on `Facter::Util::Resolvers::Http` and expect that spy to `have_received(:get_request)`